src/pyhf/pdf.py

```diff
@@ -383,6 +383,12 @@ def _expected_data(self, pars):
         allfac = tensorlib.concatenate(factors + [nom_plus_delta])
 
         newbysample = tensorlib.product(allfac, axis=0)
+        if return_by_sample:
+            batchfirst = tensorlib.swapaxes(newbysample, 0, 1)
```

Comment:
what does this do

Reply:
the array dimensions are (sample,batch,bins), while what we want to return is (batch,sample,bins) and in case of batch=None (sample,bins)

Reply:
that's why i'm confused. i thought `batch` was always first dimension...